### ISSUE TITLE
"Found an unexpected Mach-O header code: 0x72613c21" when use both XCFramework and SPM

### ISSUE THREAD

**Issue by gongzhang** [microsoft/appcenter-sdk-apple#2249]
### **Description**

When I try to reference AppCenter's **XCFramework** binary through **Swift Package Manager**, I encounter the _"Found an unexpected Mach-O header code: 0x72613c21"_ issue.

This is very similar to the other two previous issues, https://github.com/microsoft/appcenter-sdk-apple/issues/2224 and https://github.com/microsoft/appcenter-sdk-apple/issues/2200. They are related to embedded frameworks. Set the "Do not Embed" option usually solve the problem. However, when using SPM to reference XCFramework, there seems to be no similar way to set this option.

### **Repro Steps**

The steps used to reproduce this issue.

1. Create an new iOS project in latest Xcode 12.3.
2. Create a `AppCenter` subfolder in the project and add an empty `Package.swift` file.
3. Paste `Package.swift` content: [Package.swift](https://github.com/gongzhang/AppCenterXCFrameworkDemo/blob/main/AppCenter/Package.swift), which references AppCenter's XCFramework binary in SPM. ([wwdc2020/10147](https://developer.apple.com/videos/play/wwdc2020/10147/))
4. Restart Xcode and wait it downloads the binary. You should see a new "AppCenter" scheme appears if download success.
5. In setting page of main app target, add our "AppCenter" library to "Frameworks, Libraries, and Embedded Contents".
6. Archive the main app target.
7. When attemp to upload the archive in organizer, the wizard will display error message "Found an unexpected Mach-O header code: 0x72613c21".

Or, just clone [this repo](https://github.com/gongzhang/AppCenterXCFrameworkDemo), and start from step 6.

### **Discussion**

Since there is no issue when I reference [Realm's XCFramework](https://github.com/realm/realm-cocoa/releases/download/v10.5.0/realm-swift-10.5.0.zip) in the exact same way, I think there should be no problem with this kind of dependency. Please help me to see what causes the XCFramework of AppCenter not to be referenced in this way.

### **Details**

1. Which SDK version are you using?
    - 4.1.0
2. Which OS version did you experience the issue on?
    - Xcode 12.3
3. Which Cocoapods version are you using (run `pod --version`)?
    - NA
4. What device version did you see this error on?  Were you using an emulator or a physical device?
    - NA
5. What language are you using?
    - Swift
6. What third party libraries are you using?
    - NA
7. Please enable verbose logging for your app and include the logs here:
    - NA

**Comment by DmitriyKirakosyan** [comment]
Hi @gongzhang ,

Thank you for getting in touch with us!

May I ask why you are trying to create a Swift Package referencing our XCFramework?

Here is the [Docs](https://docs.microsoft.com/en-us/appcenter/sdk/getting-started/ios#33-integration-via-swift-package-manager) on how to integrate AppCenter using Swift Package Manager.

Distributing AppCenter's XCFramework via SPM is not supported.

**Comment by gongzhang** [comment]
@DmitriyKirakosyan Thanks for the reply! I've been using Swift Package Manager (SPM) to integrate with AppCenter, and the reason why I'm considering using a binary framework is because I want to **improve the compilation speed during app development**.

In particular, when using the **SwiftUI Preview** feature in Xcode, every time I perform Run/Debug command it causes all the dependencies within SPM to **recompile**. And after debugging, when I resume SwiftUI Preview, all SPM dependencies need to be **recompiled again**... This process is very time consuming, especially when the package contains a large amount of code.

Then I noticed WWDC2020 had a dedicated session (wwdc2020/10147) to introduce the topic of SPM+XCFramework, so I made this attempt. By replacing the source code with binary frameworks, this process is significantly accelerated! And I can continue to use SPM without the need to introduce other package management tools.

If it is true that for some specific reason AppCenter cannot be integrated in this way, I can continue to use the normal SPM integration method. After all, AppCenter is still a relatively lightweight library 😊.
